brave/brave-core
# Implement out of band updates for Brave Ads

Resolves https://github.com/brave/brave-browser/issues/9147

## Submitter Checklist:

- Verified that these changes build without errors on
- Verified that these changes pass automated tests (unit, browser, security tests) on

## Test Plan:

[1] Test component "Brave User Model Installer" is updating correctly
- launch < build > using `--use-dev-goupdater-url` (you should remove this arg to test production)
- check brave://components and make sure the correct component version is installed :  `Brave User Model Installer (GB) - Version: 1.0.12`
- update component via jenkins job: https://ci.brave.com/view/all-extensions/job/brave-core-ext-client-model-parameters-publish-dev
- wait ~6 hours while having < build > open and running for the scheduler to trigger an update
- verify that the new version has been downloaded by  `Brave User Model Installer (GB) - Version: 1.0.13` under brave://components
- verify models are successfully loaded after disabling/enabling ads

[2] Test Purchase Intent Classifier is working with updated model files
- search for keywords that could only be contained in dynamic lists (e.g. fintech segment)
- verify that an intent signal for that segment was extracted in `clients.json` 

[3] Test Page Classifier is working with updated model files
- check if pages are classified for supported locales

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on.
